cmake/MacroHelper.cmake

```diff
@@ -144,29 +144,21 @@ endmacro()
 # =============================================================================
 # Run nocmodl to convert NMODL to C
 # =============================================================================
-macro(nocmodl_mod_to_cpp modfile_basename)
+macro(nocmodl_mod_to_cpp modfile_basename source_dir target_dir)
   set(NOCMODL_SED_EXPR "s/_reg()/_reg_()/")
   if(NOT MSVC)
     set(NOCMODL_SED_EXPR "'${NOCMODL_SED_EXPR}'")
   endif()
-  set(REMOVE_CMAKE_COMMAND "rm")
-  if(CMAKE_VERSION VERSION_LESS "3.17")
-    set(REMOVE_CMAKE_COMMAND "remove")
-  endif()
-  get_filename_component(modfile_output_dir ${PROJECT_SOURCE_DIR}/${modfile_basename}.mod DIRECTORY)
   add_custom_command(
-    OUTPUT ${PROJECT_BINARY_DIR}/${modfile_basename}.cpp
+    OUTPUT "${target_dir}/${modfile_basename}.cpp"
     COMMAND
       ${CMAKE_COMMAND} -E env "MODLUNIT=${PROJECT_BINARY_DIR}/share/nrn/lib/nrnunits.lib"
       ${NRN_NOCMODL_SANITIZER_ENVIRONMENT} $<TARGET_FILE:${NRN_CODEGENERATOR_TARGET}>
-      ${PROJECT_SOURCE_DIR}/${modfile_basename}.mod ${NRN_NMODL_--neuron} -o ${modfile_output_dir}
-    COMMAND sed ${NOCMODL_SED_EXPR} ${PROJECT_SOURCE_DIR}/${modfile_basename}.cpp >
-            ${PROJECT_BINARY_DIR}/${modfile_basename}.cpp
-    COMMAND ${CMAKE_COMMAND} -E ${REMOVE_CMAKE_COMMAND}
-            ${PROJECT_SOURCE_DIR}/${modfile_basename}.cpp
-    DEPENDS ${NRN_CODEGENERATOR_TARGET} ${PROJECT_SOURCE_DIR}/${modfile_basename}.mod
-            ${PROJECT_BINARY_DIR}/share/nrn/lib/nrnunits.lib
-    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/src/nrniv)
+      "${source_dir}/${modfile_basename}.mod" ${NRN_NMODL_--neuron} -o "${target_dir}"
+    COMMAND sed -i -e ${NOCMODL_SED_EXPR} "${target_dir}/${modfile_basename}.cpp"
```

Comment:
I vaguely remember macs having special demands about `-i`.

Reply:
You seems to be true

Reply:
I like that it moves generated files into the build directory, we could bypass the issue by copying to `$build_dir/$mod.cpp.tmp` and then use `sed` to create `$build_dir/$mod.cpp`.

Reply:
When I did this patch the first time I vaguely remember that doing `-i -e` was how I fixed it.